crates/secrets-nats-kv/src/api.rs

```diff
@@ -0,0 +1,660 @@
+                    let secret_req: SecretRequest = match serde_json::from_slice(&payload) {
+                        Ok(r) => r,
+                        Err(_) => {
+                            let _ = self
+                                .client
+                                .publish(
+                                    reply,
+                                    SecretResponse::from(GetSecretError::InvalidRequest).into(),
+                                )
+                                .await;
+                            continue;
+                        }
+                    };
+
+                    let response = self.get(secret_req).await;
+                    match response {
+                        Ok(resp) => {
+                            let encoded = serde_json::to_vec(&resp).unwrap();
+                            let encryption_key = XKey::new();
+                            let encrypted = match encryption_key.seal(&encoded, &k) {
+                                Ok(e) => e,
+                                Err(_e) => {
+                                    let _ = self
+                                        .client
+                                        .publish(
+                                            reply,
+                                            SecretResponse::from(GetSecretError::EncryptionError)
+                                                .into(),
+                                        )
+                                        .await;
+                                    continue;
+                                }
+                            };
+
+                            let mut headers = async_nats::HeaderMap::new();
+                            headers.insert(RESPONSE_XKEY, encryption_key.public_key().as_str());
+
+                            let _ = self
+                                .client
+                                .publish_with_headers(reply, headers, encrypted.into())
+                                .await;
+                        }
+                        Err(e) => {
+                            let _ = self.client.publish(reply, e.to_string().into()).await;
+                        }
+                    }
+                }
+                // Custom handlers
+                // These handlers are not part of the wasmCloud secrets spec, but are provided in
+                // in order to extend the secrets backend in order to make it usable.
+                "add_mapping" => {
+                    let entity = match parts.get(OPERATION_INDEX + 1) {
+                        Some(e) => e,
+                        None => {
+                            let _ = self
+                                .client
+                                .publish(reply, "no entity provided".into())
+                                .await;
+                            continue;
+                        }
+                    };
+
+                    let payload = msg.payload;
```

Comment:
Super duper nit: I don't know if it would technically make it any more efficient or not since JSON allocates from borrowed data, but if you import the `bytes::Buf` trait, you can shorten all these to be `serde_json::from_reader(msg.payload.reader())` while consuming the payload